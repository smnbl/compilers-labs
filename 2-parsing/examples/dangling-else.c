int dangling_else()
{
    if (1)
        if (2)
            a;
        else
            b;
}
