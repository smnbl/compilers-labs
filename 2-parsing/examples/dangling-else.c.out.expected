└── Program
    └── FuncDecl: returnType = 'int', name = 'dangling_else'
        └── CompoundStmt
            └── IfStmt
                ├── IntLiteral: value = '1'
                └── IfStmt
                    ├── IntLiteral: value = '2'
                    ├── ExprStmt
                    │   └── VarRefExpr: name = 'a'
                    └── ExprStmt
                        └── VarRefExpr: name = 'b'
